confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java

```diff
@@ -218,7 +221,32 @@ public TSStatus deleteStorageGroups(TDeleteStorageGroupsReq tDeleteReq) throws T
 
   @Override
   public TSStatus setTTL(TSetTTLReq req) throws TException {
-    return configManager.setTTL(new SetTTLReq(req.getStorageGroup(), req.getTTL()));
+    if (!configManager.ifStorageGroupExist(req.storageGroup)) {
+      return RpcUtils.getStatus(
+          TSStatusCode.STORAGE_GROUP_NOT_EXIST,
+          "storageGroup " + req.storageGroup + " does not exist");
+    }
+    TSStatus tsStatus = new TSStatus();
+    tsStatus = configManager.setTTL(new SetTTLReq(req.getStorageGroup(), req.getTTL()));
+    Set<TDataNodeLocation> dataNodeLocations =
+        configManager
+            .getPartitionManager()
```

Comment:
call the getDataNodeLocatoinList in ConfigManager with confirmLeader()

We need to read/write partition information through ConfigNode manager currently.